src/transactions/transaction_builder/remoteAbi.ts

```diff
@@ -0,0 +1,251 @@
+import { MoveFunction } from "../../types";
+import {
+  parseTypeTag,
+  TypeTag,
+  TypeTagAddress,
+  TypeTagBool,
+  TypeTagStruct,
+  TypeTagU128,
+  TypeTagU16,
+  TypeTagU256,
+  TypeTagU32,
+  TypeTagU64,
+  TypeTagU8,
+  TypeTagVector,
+} from "../typeTag";
+import { AptosConfig } from "../../api/aptosConfig";
+import { EntryFunctionArgumentTypes, SimpleEntryFunctionArgumentTypes } from "../types";
+import { Bool, FixedBytes, MoveOption, MoveString, MoveVector, U128, U16, U256, U32, U64, U8 } from "../../bcs";
+import { AccountAddress, Hex } from "../../core";
+import { getModule } from "../../internal/account";
+
+/**
+ * Finds first non-signer arg.
+ *
+ * A function is often defined with a `signer` or `&signer` arguments at the start, which are filled in
+ * by signatures, and not by the caller.
+ * @param functionAbi
+ */
+export function findFirstNonSignerArg(functionAbi: MoveFunction): number {
+  return functionAbi.params.findIndex((param) => param !== "signer" && param !== "&signer");
+}
+
+/**
+ * Common logic for throwing when there are too many arguments
+ * @param functionId
+ * @param argumentIndex
+ * @param numExpected
+ */
+export function ensureNotTooManyArguments(functionId: string, argumentIndex: number, numExpected: number) {
+  if (argumentIndex >= numExpected) {
+    throw new Error(`Too many arguments for '${functionId}', expected ${numExpected}`);
+  }
+}
+
+/**
+ * Convert type arguments to only type tags, allowing for string representations of type tags
+ */
+export function typeTagConversion(typeArguments?: Array<TypeTag | string>): Array<TypeTag> {
+  return (
+    typeArguments?.map((typeArg: string | TypeTag): TypeTag => {
+      // Convert to TypeTag if it's a string representation
+      if (typeof typeArg === "string") {
+        return parseTypeTag(typeArg);
+      }
+      return typeArg;
+    }) ?? []
+  );
+}
+
+/**
+ * Fetches the ABI for an entry function from the module
+ * @param moduleAddress
+ * @param moduleName
+ * @param functionName
+ * @param aptosConfig
+ */
+export async function fetchEntryFunctionAbi(
+  moduleAddress: string,
+  moduleName: string,
+  functionName: string,
+  aptosConfig: AptosConfig,
+) {
+  const module = await getModule({ aptosConfig, accountAddress: moduleAddress, moduleName });
+
+  // TODO: Memoize? or is the module enough
```

Comment:
What is the caching mechanim on this?

Reply:
It's cached once every 5 minutes, but I'm changing the way that the caching works to be after the ABI is parsed rather than before.

Reply:
I wonder if we can just cache for longer by default + provide some advanced way to change it. Contract upgrades are very infrequent and users can always refresh the UI or restart a job?